### DISCUSSION TITLE
Singapore OVH server not able to connect to Github

### DISCUSSION THREAD

**Discussion by camgullo** [community/community#40172] [New to GitHub]
Hello,

I have an OVH Server in Singapore which is not responding to wget or ping requests to Github. 

[root@sgp-vz1 ~]# ping github.com
PING github.com (20.205.243.166) 56(84) bytes of data.
^C
--- github.com ping statistics ---
7 packets transmitted, 0 received, 100% packet loss, time 6001ms

When I ping "20.205.243.166" on my local connection, it also doesn't respond,
Camerons-MacBook-Pro:~ camerongullo$ ping 20.205.243.166
PING 20.205.243.166 (20.205.243.166): 56 data bytes
Request timeout for icmp_seq 0
Request timeout for icmp_seq 1
Request timeout for icmp_seq 2
^C
--- 20.205.243.166 ping statistics ---
4 packets transmitted, 0 packets received, 100.0% packet loss

On my local connection, If I ping github.com it responds with,
Camerons-MacBook-Pro:~ camerongullo$ ping github.com
PING github.com (140.82.112.4): 56 data bytes
64 bytes from 140.82.112.4: icmp_seq=0 ttl=49 time=28.858 ms
64 bytes from 140.82.112.4: icmp_seq=1 ttl=49 time=28.159 ms
64 bytes from 140.82.112.4: icmp_seq=2 ttl=49 time=24.546 ms
^C
--- github.com ping statistics ---
3 packets transmitted, 3 packets received, 0.0% packet loss
round-trip min/avg/max/stddev = 24.546/27.188/28.858/1.890 ms

When I ping "140.82.112.4" on the OVH server, it responds,
[root@sgp-vz1 ~]# ping 140.82.112.4
PING 140.82.112.4 (140.82.112.4) 56(84) bytes of data.
64 bytes from 140.82.112.4: icmp_seq=1 ttl=48 time=228 ms
64 bytes from 140.82.112.4: icmp_seq=2 ttl=48 time=228 ms
64 bytes from 140.82.112.4: icmp_seq=3 ttl=48 time=227 ms
^C
--- 140.82.112.4 ping statistics ---
3 packets transmitted, 3 received, 0% packet loss, time 2002ms
rtt min/avg/max/mdev = 227.550/228.217/228.571/0.726 ms

I'm not too sure what is causing this, if anyone has any suggestions I would love to hear them!

Thanks,
Cam

**Comment by miyurusankalpa** [comment]
https://github-debug.com/

**Reply by camgullo** [reply]
Thanks for that, I just submitted a ticket. I will keep this thread updated when I get more information.

Thanks,
Cam

**Reply by camgullo** [reply]
In regard to your email,
> GitHub does GEODNS and they have a local azure node in SG.
> https://dns.google/query?name=github.com&rr_type=A&ecs=2400%3A6500%3A%3A2a
> I am currently getting 20.205.243.166
>
> If you can reach it, you might want to check your DNS resolver and it will solve the issue temporarily.

So I do reach, 20.205.243.166 however it doesn't respond on my end. Does it respond for you?

I try clearing the DNS cache using, systemctl restart dnsmasq.service
I tried changing the server’s IP to a new IP, all to no avail.

Thanks,
Cam

**Reply by miyurusankalpa** [reply]
Sorry about that, I saw the IP 20.205.243.166 in your question just after posting that. So I deleted it since its not helpful to debug the issue.

and yes, 20.205.243.166 works for me(LK ISP).

**Reply by camgullo** [reply]
Really, That IP responds for you? I'm surprised, Looking at, https://i.ping.pe/D/A/img_DAoN6Wpr.png
It shows 100% loss throughout the world. 

And doing a [GEO DNS lookup](https://www.whatsmydns.net/#A/github.com) the IP “ 20.205.243.166” is shown for Singapore.

**Reply by miyurusankalpa** [reply]
They seem to be blocking ICMP but HTTP works.

CURL: `* Connected to github.com (20.205.243.166) port 443 (#0)`

Browser:
![image](https://user-images.githubusercontent.com/1833185/204876523-3eda0f49-db7b-4322-876e-bb8c76a6f007.png)
